plugin/vim-ai.vim

```diff
@@ -4,8 +4,25 @@ if !has('python3')
   finish
 endif
 
-command! -range -nargs=? AI <line1>,<line2>call vim_ai#AIRun(<range>, {}, <f-args>)
-command! -range -nargs=? AIEdit <line1>,<line2>call vim_ai#AIEditRun(<range>, {}, <f-args>)
-command! -range -nargs=? AIChat <line1>,<line2>call vim_ai#AIChatRun(<range>, {}, <f-args>)
+" detect if a visual selection is pending
+augroup vim_ai
+  autocmd!
+  if exists('##ModeChanged')
+    autocmd ModeChanged         *:[vV\x16]* let g:vim_ai_is_selection_pending = 1
+    autocmd ModeChanged [vV\x16]*:*
+          \ let g:vim_ai_is_selection_pending = v:event.new_mode =~# '^c' && getcmdline() =~# "^\s*'<'>"
+  else
```

Comment:
using `ModeChanges` looks like an elegant solution, do we still need code in `else` branch?

Reply:
I'd wage yes, since it came with 9.0 (more exactly [8.2.3424](https://github.com/vim/vim/commit/f14b8ba1373f569705cb80419248054100b02360) according to [vimlog](https://www.arp242.net/vimlog/)) and till now your plug-in works about everywhere (since the openai package dependency is gone).